src/NLog/Config/XmlLoggingConfiguration.cs

```diff
@@ -321,27 +325,24 @@ private void Initialize(XmlReader reader, string fileName, bool ignoreErrors)
             catch (Exception exception)
             {
                 InitializeSucceeded = false;
-                if (exception.MustBeRethrown())
+
+                if (exception.IsServereException())
                 {
                     throw;
                 }
 
-                NLogConfigurationException ConfigException = new NLogConfigurationException("Exception occurred when loading configuration from " + fileName, exception);
+                var message = string.Format("Exception occurred when loading configuration from '{0}': {1}", fileName, exception);
 
-                if (!ignoreErrors)
+                if (ignoreErrors)
                 {
-                    if (LogManager.ThrowExceptions)
-                    {
-                        throw ConfigException;
-                    }
-                    else
-                    {
-                        InternalLogger.Error("Error in Parsing Configuration File. Exception : {0}", ConfigException);
-                    }
+                    InternalLogger.Error(message);
                 }
                 else
                 {
-                    InternalLogger.Error("Error in Parsing Configuration File. Exception : {0}", ConfigException);
+                    if (exception.MustBeRethrown(message))
+                    {
+                        throw new NLogConfigurationException(message);
+                    }
                 }
             }
         }
```

Comment:
This case is tricky and it is forcing us to duplicate some of the exception-handling logic in `MustBeRethrown()` (eg. when must an exception be rethrown and how we log exceptions). How about adding more parameters to `HandleException()` to accommodate this?

My suggestion is based on the previous:
> If we go with HandleException() I would suggest the method be void and it does the logging and then rethrows the exception itself.

In order to accommodate the case above, we could add the following parameters:
* `wrappedExceptionType` - eg. `typeof(NLogConfigurationException)`; use `null` if the exception must not be wrapped.
* `ignoreErrors` - by default `false`; if `true` we won't rethrow unless the exception is severe/fatal.

Perhaps we should wrap all the parameters as properties in a new `HandleExceptionOptions` class with appropriate default values.